src/InsertCalorimeter_geo.cpp

```diff
@@ -49,7 +49,7 @@ static Ref_t createDetector(Detector& desc, xml_h handle, SensitiveDetector sens
   const std::pair<double, double> hole_y_parameters(hole_y_initial, hole_y_final);
 
   // Getting thickness of backplate
-  auto backplate_thickness = dd4hep::getAttrOrDefault<double>(detElem, _Unicode(backplate_thickness), 0. * cm);
+  auto backplate_thickness = detElem.hasChild(_Unicode(backplate))? detElem.child(_Unicode(backplate)).attr<double>(_Unicode(thickness)) : 0.;
```

Comment:
This looks a bit janky (ternary operator is not very readable). I assume there's somewhere else another `if` to then place the backplate or not.

Reply:
Ah, no there's not actually. The backplate is included in the layer section that's placed via the existing for loop. But I need the thickness of the final layer (which is the backplate) before the for loop to make a function that calculates the radius and position of the beampipe hole.

Reply:
Err if that wasn't clear, the hole shape is determined by a linear interpolation from the front of the insert to the front of the final layer. So I need the (total length - final layer length) to calculate the hole info.

Reply:
Nah, this is fine. Maybe just add a comment here to clarify.